plugins/modules/volume_cloudinit.py

```diff
@@ -130,7 +130,11 @@ def cloud_localds(volume_format,
     #       because it is not available on Red Hat Enterprise Linux 8 / 9 and CentOS 8 / 9.
     #       Ref.: https://salsa.debian.org/cloud-team/cloud-utils/-/blob/master/bin/cloud-localds
 
-    cmd = 'cloud-localds'
+    
+    cmd = "cloud-localds"
+    check_sudo_access =  module.run_command('which cloud-localds')
+    if check_sudo_access is not "/usr/local/sbin/cloud-localds":
+        cmd = '/usr/local/sbin/cloud-localds'
```

Comment:
```bash
$ sudo which cloud-localds
which: no cloud-localds in (/sbin:/bin:/usr/sbin:/usr/bin)
]$ which cloud-localds
/usr/local/sbin/cloud-localds
```

Reply:
You might be looking at this problem from the wrong direction?! Why is sudo not looking in `/usr/local/sbin` on your sys?

Reply:
> You might be looking at this problem from the wrong direction?! Why is sudo not looking in `/usr/local/sbin` on your sys?

hmm seem to be an issue with some distros though:
https://www.linuxquestions.org/questions/debian-26/why-is-usr-sbin-not-in-root%27s-path-4175736161/
not the same but similar:
https://bugzilla.redhat.com/show_bug.cgi?id=587183

Reply:
Both are unrelated to your problem, it seems. First one is Debian, second one is a different path. Are you running RHEL9 in a container or in a VM or on bare-metal?

Reply:
> Are you running RHEL9 in a container or in a VM or on bare-metal?
- bare metal rocky linux 9.4 machine

i think this might be related to puppet, which i had installed before.
puppet & tailscale is the only thing i have installed on this machine besides stuff like wget. thats why this is so weird, because those machines are freshly provisioned.
but i will test on my other server which just has tailscale installed.

Reply:
so yeah this can be related to puppet:
https://www.puppet.com/docs/puppet/7/environments_creating.html
codeium says  that puppet environment resource can overwrite your environment path like so:
```
user { 'root':
  environment => {
    'PATH' => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
  },
}
```

Reply:
I am not familiar with Puppet, but the `PATH` var listed above would work for this role, too. Your issue probably is somewhere else.

Reply:
you can close this pr now. i wrote me a workflow that automatically creates hosts from prototypes including random mac, passwd etc... i also figured out how to automate and secure my ssh using vault.  i then store those facts, update  vault and run your collection. so everything works fine now. 
thanks bro!

Reply:
Ok, glad to hear that you found a way that works for you :)